### PR TITLE
Remove deprecated code

### PR DESCRIPTION
**Pull request by pchaigno**
This pull request removes deprecated code. See commit descriptions for details.

### CODE REVIEW

**Comment by pchaigno** [comment]
test-me-please

**Comment by pchaigno** [comment]
test-1.19-5.4

**Comment by pchaigno** [comment]
Blocked on https://github.com/cilium/cilium/pull/16483.

**Comment by pchaigno** [comment]
test-me-please

**Comment by pchaigno** [comment]
test-me-please

File: pkg/option/config.go [hunk 1]
```diff
@@ -2736,16 +2707,11 @@ func (c *DaemonConfig) Populate() {
...
 
 func (c *DaemonConfig) populateMasqueradingSettings() error {
 	switch {
-	case viper.IsSet(Masquerade) && viper.IsSet(EnableIPv4Masquerade):
-		return fmt.Errorf("--%s and --%s (deprecated) are mutually exclusive", EnableIPv4Masquerade, Masquerade)
-	case viper.IsSet(Masquerade):
-		c.EnableIPv4Masquerade = viper.GetBool(Masquerade) && c.EnableIPv4
 	default:
 		c.EnableIPv4Masquerade = viper.GetBool(EnableIPv4Masquerade) && c.EnableIPv4
 	}
```
Comment (christarazi) [inline pkg/option/config.go]: No need for this switch statement at all then :rocket:

**Comment by pchaigno** [comment]
test-me-please

**Comment by pchaigno** [comment]
For some reason, GitHub's load balancing of reviews didn't work on this pull request so I manually assigned some folks (while trying to minimize the total number of required reviewers).

**Review by qmonnet — APPROVED** [review]

**Review by tklauser — APPROVED** [review]

**Review by christarazi — APPROVED** [review]

**Comment by pchaigno** [comment]
All team review requests are covered. Tests are passing except for ConformanceAKS which is very unstable and not required. Marking as ready to merge.

**Review by aditighag — APPROVED** [review]

**Pull request merged**